src/nvim/event/wstream.c

```diff
@@ -73,6 +73,26 @@ bool wstream_write(Stream *stream, WBuffer *buffer)
   // This should not be called after a stream was freed
   assert(!stream->closed);
 
+  uv_buf_t uvbuf;
+  uvbuf.base = buffer->data;
+  uvbuf.len = UV_BUF_LEN(buffer->size);
+
+  if (!stream->uvstream) {
+    uv_fs_t req;
+
+    // Synchronous write
+    uv_fs_write(stream->uv.idle.loop, &req, stream->fd, &uvbuf, 1, stream->fpos, NULL);
+
+    uv_fs_req_cleanup(&req);
+
+    wstream_release_wbuffer(buffer);
+
+    assert(stream->write_cb == NULL);
```

Comment:
should this go just after the if condition (line 80)?

Reply:
I was trying to match the order in the `write_cb()` below, where `write_cb` is checked after calling `wstream_release_buffer()`, and `closed` is checked afterwards. When `write_cb` is `NULL`, `closed` won't be set here, so `wstream_release_wbuffer()` is the only one needed here. An alternative is to move that part of `write_cb()` into a separate function, but then the `pending_reqs--` needs to be paired with a `pending_reqs++`, which I'm not sure where to put in this branch.